### DISCUSSION TITLE
Can I call a model method on form dehydration?

### DISCUSSION THREAD

**Discussion by emincer** [filamentphp/filament#6543] [Help] [answered]
In my table listing I use a method from the model such as this:

`                Tables\Columns\TextColumn::make('getmostRecentToken.balance')->label("Tokens"),
`

This function returns a token count in the user model. Naturally I'd like to set it in the edit form.  I already have a method in the Model for updating a user token count:

`    public function adjustTokenBalance($event, $amount, $balance = NULL)
`

Is there any way to connect that to my form?

**Comment by danharrin** [comment, marked as answer]
You could use an action maybe? https://filamentphp.com/docs/2.x/admin/pages/actions

But otherwise, you could use a model observer to call the `adjustTokenBalance()` automatically if the field is changed
